src/main/java/nwoolcan/model/brewery/warehouse/article/ArticleManager.java

```diff
@@ -0,0 +1,137 @@
+package nwoolcan.model.brewery.warehouse.article;
+
+import nwoolcan.model.utils.UnitOfMeasure;
+import nwoolcan.utils.Result;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Manager for {@link Article} objects. It is used by tests and by the
+ * {@link nwoolcan.model.brewery.warehouse.Warehouse} in order to create {@link Article}, check the
+ * id of the {@link Article}, avoid repetitions and set name of the {@link Article}.
+ */
+public final class ArticleManager {
+
+    @Nullable private static ArticleManager instance;
+    private static final String ARTICLE_NOT_REGISTERED = "The article was not registered. You can not change its name.";
+    private static final String ARTICLE_WITH_NEW_NAME_ALREADY_REGISTERED = "Changing the name to this article would produce an article which already exists.";
+    private static int fakeId = -1;
+    private int nextAvailableId;
+    private Map<Article, Integer> articleToId;
+    private Map<Integer, Article> idToArticle;
```

Comment:
Dont really understand why having two maps.
Can you explain?

Reply:
explained in the other question, following the thread